prow/initupload/run.go

```diff
@@ -22,25 +22,49 @@ import (
 	"errors"
 	"fmt"
 	"io/ioutil"
+	"strconv"
 	"time"
 
 	"k8s.io/test-infra/prow/pod-utils/clone"
 	"k8s.io/test-infra/prow/pod-utils/downwardapi"
 	"k8s.io/test-infra/prow/pod-utils/gcs"
 )
 
+// specToStarted translate a jobspec into a started struct
+func specToStarted(spec *downwardapi.JobSpec) gcs.Started {
+	started := gcs.Started{
+		Timestamp:   time.Now().Unix(),
+		RepoVersion: downwardapi.GetRevisionFromSpec(spec),
```

Comment:
Looks like this should be RepoCommit, rather than RepoVersion?

https://github.com/kubernetes/test-infra/blob/9bb09552a4317405d8b9bad9be770ea154fdbfe9/jenkins/bootstrap.py#L503

Looks like I goofed here:

https://github.com/kubernetes/test-infra/blob/60bca00c14b62d24af333aa3e160dc56368679ac/testgrid/metadata/job.go#L31

And checking testgrid's internal updater it has been looking for repo-commit, job-version and/or version since 2017. So we should use repo-commit, not repo-version.

which somewhat makes sense, as repo-commit is a git commit, whereas job-version is a random string set by the job.

Reply:
done

Reply:
hold on hold on - this is not true at all - see https://github.com/kubernetes/test-infra/blob/aa8d6d1c02a9deb17f3c631a4ac1dc87885fbdb7/jenkins/bootstrap.py#L322, repo-commit is a field in metadata in finished.json instead of started.json

although this repo-version in started.json is :-1: (https://github.com/kubernetes/test-infra/blob/aa8d6d1c02a9deb17f3c631a4ac1dc87885fbdb7/jenkins/bootstrap.py#L596-L621) - so - I guess I can just do a rev-parse like https://github.com/kubernetes/test-infra/pull/11330?

Reply:
Should link this thread to my "WTF are these fields even" issue :laughing: